data-prepper-api/src/main/java/org/opensearch/dataprepper/typeconverter/StringConverter.java

```diff
@@ -25,6 +34,11 @@ public String convert(Object source) throws IllegalArgumentException {
         if (source instanceof String) {
             return (String)source;
         }
-        throw new IllegalArgumentException("Unsupported type conversion");
+        LOG.error(EVENT, "Unable to convert {} to String", source);
+        if (Objects.nonNull(source)) {
+            return source.toString();
+        } else {
+            return "";
```

Comment:
In the case of `null`, we can probably handle this in the `ConvertEntryTypeProcessor` class. Any `null` value can be converted to `null` for the target.

Also, it should not be an empty string. It should be `null`.

Reply:
I can return `null` string in case of null value instead of empty string. convert method is also used in translate processor.